src/assessment/views.py

```diff
@@ -157,10 +157,22 @@ class AffectDetailView(LoginRequiredMixin, TemplateView):
     """
     student_template_name = 'shared/partial/modal_affect_detail.html'
     teacher_template_name = 'shared/partial/modal_class_affect_detail.html'
+    student_details_template_name = 'shared/partial/modal_student_details_affect.html'
 
     def get(self, request, *args, **kwargs):
         self.word = kwargs['word']
-        clusive_user = request.clusive_user
+
+        # If a username is provided by kwargs['for_user'], set the clusive_user
+        # to that user. If there is no such user, or no for_user in kwargs, use
+        # the request.clusive_user
+        self.for_user = kwargs.get('for_user')
+        try:
```

Comment:
I wonder if, in the None case, this will actually query the database for ClusiveUser "where username is null".  Would be cleaner to handle that case with an if.

Reply:
Okay, done.